src/Listeners/UpdateSyncedResource.php

```diff
@@ -112,12 +113,28 @@ protected function updateResourceInTenantDatabases($tenants, $event, $syncedAttr
                 if ($localModel) {
                     $localModel->update($syncedAttributes);
                 } else {
-                    // When creating, we use all columns, not just the synced ones.
-                    $localModel = $localModelClass::create($eventModel->getAttributes());
+                    $localModel = $localModelClass::create($this->getAttributesForCreation($eventModel));
```

Comment:
Could `$eventModel` come from the central database here? If so, then I think according to what we discussed here https://github.com/archtechx/tenancy/pull/915#issuecomment-1255807197, it would take the `getSyncedCreationAttributes()` from the **incoming** model rather than the local one that we're trying to create?

Do we have tests to ensure that the `getSyncedCreationAttributes()` are taken from the right model?

Reply:
I'm a little confused here.

> it would take the `getSyncedCreationAttributes()` from the incoming model rather than the local one that we're trying to create?

If I understand correctly, You are saying that `getSyncedCreationAttributes()` are coming from the incoming model instead local model. 

Yes, it is. Why should it come from the "local one we're trying to create"? I mean, we are copying one model to another, So we are asking the resource model (which is being copied) to provide the attributes for the target model (the model will be created). 

> Do we have tests to ensure that the getSyncedCreationAttributes() are taken from the right model?

We don't have a specific test for this. Don't you think [current tests](https://github.com/archtechx/tenancy/pull/915/files#diff-0c32d140e95604d5ab8852beb5eba2255180184351adfaa0ff7b2895012d4d7dR130-R309) cover that `getSyncedCreationAttributes()` are taken from the right model?

Reply:
> Why should it come from the "local one we're trying to create"?

1. We're creating the local model
2. The synced attributes include a list of attributes that are used, including defaults
3. We want to use defaults for the local model, not defaults for the incoming model

I'm not 100% sure about this, but it seems wrong to use `getSyncedCreationAttributes()` from a different model than the one we're creating.

Read our previous conversation: https://github.com/archtechx/tenancy/pull/915#issuecomment-1255807197

I specifically asked if there can be different `getSyncedCreationAttributes()` on the central and tenant model.

Reply:
Or do you think that the `getSyncedCreationAttributes()` **should always be called on the opposite model** than the one that we're creating?

Reply:
> Or do you think that the getSyncedCreationAttributes() should always be called on the opposite model than the one that we're creating?

Yes, that is what I see and also code doing right now. Do you want me to write a detailed message about the implementation of BC?

Reply:
Based on the BC discussion, a comment has been added so resolve the review.